src/common/post.ts

```diff
@@ -534,3 +539,87 @@ export const processApprovedModeratedPost = async (
 
   return null;
 };
+
+export const validateSourcePostModeration = async (
+  ctx: AuthContext,
+  {
+    postId,
+    title,
+    content,
+    sourceId,
+    image,
+    type,
+    sharedPostId,
+    imageUrl,
+    externalLink,
+  }: CreateSourcePostModerationArgs,
+): Promise<CreateSourcePostModeration> => {
+  if (![PostType.Share, PostType.Freeform].includes(type)) {
+    throw new ValidationError('Invalid post type!');
+  }
+
+  const { con, userId } = ctx;
+
+  const sourceMember = await con.getRepository(SourceMember).findOne({
+    where: { sourceId: sourceId, userId: userId },
+    relations: ['source'],
+  });
+
+  const source = await sourceMember?.source;
+
+  if (!source || source.type !== SourceType.Squad) {
+    throw new ForbiddenError('Access denied!');
+  }
+
+  await canAccessSource(
+    ctx,
+    source,
+    sourceMember,
+    SourcePermissions.Post,
+    sourceId,
+  );
+
+  const pendingPost: CreateSourcePostModeration = {
+    title: validateCommentary(title),
+    postId,
+    sourceId,
+    type,
+    sharedPostId,
+    externalLink,
+    createdById: userId,
+  };
+
+  const mentions = await getMentions(con, content, userId, sourceId);
+
+  if (type === PostType.Share) {
+    if (!!externalLink) {
+      const cleanContent = validateCommentary(content);
+
+      if (cleanContent) {
+        pendingPost.content = cleanContent;
+        pendingPost.contentHtml = generateTitleHtml(cleanContent, mentions);
+      }
+    } else if (pendingPost.title) {
+      pendingPost.titleHtml = generateTitleHtml(pendingPost.title, mentions);
+    }
```

Comment:
else only?

Reply:
Was thinking about that, but we wouldn't want to process anything if the title is empty.